File: backend/semgrep_rules/tests/java/kisa_sw_35_infinite_loop.java
class kisa_sw_35_infinite_loop {
    void unsafe() {
        // ruleid: kisa.sw35.java.obvious-infinite-loop
        while (true) {
            System.out.println("loop");
        }
    }
    void safe(int n) {
        // ok: kisa.sw35.java.obvious-infinite-loop
        for (int i = 0; i < n; i++) System.out.println(i);
    }
}

File: backend/semgrep_rules/tests/javascript/kisa_sw_35_infinite_loop.js
function unsafe() {
  // ruleid: kisa.sw35.javascript.obvious-infinite-loop
  while (true) {
    console.log("loop");
  }
}
function safe(n) {
  // ok: kisa.sw35.javascript.obvious-infinite-loop
  for (let i = 0; i < n; i++) console.log(i);
}
